pkg/reconciler/openshift/tektonconfig/extension/openshift_cleanup.go

```diff
@@ -0,0 +1,42 @@
+package extension
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/tektoncd/operator/pkg/reconciler/common"
+
+	mf "github.com/manifestival/manifestival"
+	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
+	"knative.dev/pkg/logging"
+)
+
+func AppendCleanupTarget(ctx context.Context, manifest *mf.Manifest, instance v1alpha1.TektonComponent) error {
+	manifestPath := filepath.Join(common.ComponentDir(instance), "99-clean-up")
+	m, err := common.Fetch(manifestPath)
+	if err != nil {
+		return err
+	}
+	*manifest = manifest.Append((m))
+	return nil
+}
+
+func CleanupTransforms(ctx context.Context, manifest *mf.Manifest, instance v1alpha1.TektonComponent) error {
+	return common.Transform(ctx, manifest, instance)
+}
+
+func RunCleanup(ctx context.Context, manifest *mf.Manifest, instance v1alpha1.TektonComponent) error {
+	logger := logging.FromContext(ctx)
+	logger.Debug("Running Cleanup Jobs on OpenShift")
+	status := instance.GetStatus()
+	if err := manifest.Apply(); err != nil {
+		status.MarkInstallFailed(err.Error())
+		return fmt.Errorf("failed to apply cleanup job: %w", err)
+	}
+	return nil
+}
+
+func CheckCleanup(ctx context.Context, manifest *mf.Manifest, instance v1alpha1.TektonComponent) error {
+	return nil
```

Comment:
As its returning nil do we require this function ?

Reply:
i kepth it as a place holder to implement the logic to check whether the job was successful.
I should have added a  `TODO:` comment.

Shall I handle it in my next pr.

Reply:
Sure :+1: